### PR TITLE
CYF-HOTEL-PROJECT-TAYFUN

### PR DESCRIPTION
**Pull request by tayfuntayfun**

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+  "cSpell.ignoreWords": ["noopener", "noreferrer", "peoplemakeglasgow"]
+}
```
Comment (mozartdiniz) [inline .vscode/settings.json:3]: please don't submit your IDE configuration in this PR. You can always add it to .gitignore

File: src/App.js [hunk 2]
```diff
@@ -1,13 +1,23 @@
...
   return (
     <div className="App">
-      <header className="App-header">CYF Hotel</header>
+      <Heading />
+      <TouristInfoCards />
       <Bookings />
+      <Footer countryInfo={countryInfo} />
     </div>
   );
 };
```
Comment (mozartdiniz) [inline src/App.js:23]: Nice job in this App component. Simple and clean.

File: src/Bookings.js [hunk 3]
```diff
@@ -1,7 +1,7 @@
 import React from "react";
 import Search from "./Search.js";
-// import SearchResults from "./SearchResults.js";
-// import FakeBookings from "./data/fakeBookings.json";
+import SearchResults from "./SearchResults.js";
+import FakeBookings from "./data/fakeBookings.json";
```
Comment (mozartdiniz) [inline src/Bookings.js:4]: In JS, and when using React as well, we should only capitalize variables when it is a Class or a React Component.

File: src/Footer.js [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+    <footer className="footer">
+      <ul>
+        {props.countryInfo.map((element, index) => (
+          <li key={index}>{element}</li>
+        ))}
+      </ul>
+    </footer>
+  );
+};
+export default Footer;
```
Comment (mozartdiniz) [inline src/Footer.js:14]: Nice one!

File: src/Heading.js [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+      <img
+        className="logo-img"
+        src="https://image.flaticon.com/icons/svg/139/139899.svg"
+      />
+      <header className="App-header">CYF Hotel</header>
+    </div>
+  );
+};
+
+export default Heading;
```
Comment (mozartdiniz) [inline src/Heading.js:15]: Good job. Do you think we could receive this `CYF Hotel` via props?

File: src/SearchResults.js [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+          <th scope="col">email</th>
+          <th scope="col">room id</th>
+          <th scope="col">check in date</th>
+          <th scope="col">check out date</th>
+          <th scope="col">Nights</th>
+        </tr>
+      </thead>
+      <tbody>
+        {props.results.map((booking, index) => {
+          const checkIn = moment(booking.checkOutDate);
```
Comment (mozartdiniz) [inline src/SearchResults.js:22]: nice!

File: src/SearchResults.js [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+              <td>{booking.checkOutDate}</td>
+              <td>{checkIn.diff(checkOut, "days")}</td>
+            </tr>
+          );
+        })}
+      </tbody>
+    </table>
+  );
+};
+export default SearchResults;
```
Comment (mozartdiniz) [inline src/SearchResults.js:42]: Good job.

File: src/SearchButton.js [hunk 8]
```diff
@@ -0,0 +1,7 @@
+import React from "react";
+
+const SearchButton = () => {
+  return <button className="btn btn-primary">Search</button>;
```
Comment (mozartdiniz) [inline src/SearchButton.js:4]: Do you think we could receive this `Search` via props and make this button more generic?

File: src/TouristInfoCards.js [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+          <a
+            href="https://www.visitmanchester.com/"
+            target="_blank"
+            className="btn btn-primary"
+            rel="noopener noreferrer"
+          >
+            Manchester
+          </a>
+        </div>
+      </div>
```
Comment (mozartdiniz) [inline src/TouristInfoCards.js:37]: Don't you think we could make this `<div className="card">` a component, and then, make it receive that info via props?

Another point, it is always important to try to don't repeat code. So, in this case, what if we created a data structure which we could map and return this component? Wouldn't be cool? We could avoid copy and paste this component over and over again.

Something like:
```js
const touristInformations = [{
   city: 'Glasgow',
   site: 'https://peoplemakeglasgow.com/',
   imageURL: '"https://frontline-collections.com/wp-content/uploads/2018/02/Glasgow-Debt-Collection.jpg'
}, {
   city: 'Manchester',
   site: 'https://www.visitmanchester.com/',
   imageURL: 'https://www.prolificnorth.co.uk/sites/default/files/styles/lightbox_large/public/images/news/manchester-cityscape0.jpg?itok=VjQDdTYJ'
}];
```

What do you think?

File: .vscode/settings.json (hunk 1 above)
Comment (tayfuntayfun) [reply .vscode/settings.json:3]: Hi Mozart, thanks for this feedback. I am not quite sure if I understood this. Could you explain me how to do this?
Thanks

File: src/App.js (hunk 2 above)
Comment (tayfuntayfun) [reply src/App.js:23]: Thanks

File: src/Bookings.js (hunk 3 above)
Comment (tayfuntayfun) [reply src/Bookings.js:4]: I am still mixing this

File: src/Heading.js (hunk 5 above)
Comment (tayfuntayfun) [reply src/Heading.js:15]: Yes, we could definitely.

File: src/SearchButton.js (hunk 8 above)
Comment (tayfuntayfun) [reply src/SearchButton.js:4]: Would it make more sense then! I am not quite sure?

File: src/TouristInfoCards.js (hunk 9 above)
Comment (tayfuntayfun) [reply src/TouristInfoCards.js:37]: Great advice. I have re-constructed the code as you advised. Thanks

**Comment by 40thieves** [comment]
Not quite sure what happened here @tayfuntayfun, but it looks like you're continuing in https://github.com/CodeYourFuture/cyf-hotel-react/pull/126. I'm going to close this PR so that we don't get confused.

**Comment by tayfuntayfun** [comment]
Hi Alaistir, my #99 has had a broken Repository link and I had a chat with
Jon and he advised another PR.

**Comment by tayfuntayfun** [comment]
Yes I am continuing w #126

**Pull request closed without merging**